### PR TITLE
Fix read data out of boundary cause System.AccessViolationException

### PR DESCRIPTION
**Pull request by shaopeng-gh**

### CODE REVIEW

File: src/BinaryParsers/ElfBinary/Dwarf/DwarfMemoryReader.cs [hunk 1]
```diff
@@ -249,6 +249,11 @@ public byte[] ReadBlock(uint size)
         /// <param name="position">The position.</param>
         public byte[] ReadBlock(uint size, int position)
         {
+            if (position >= Data.Length)
```
Comment (eddynaka) [inline src/BinaryParsers/ElfBinary/Dwarf/DwarfMemoryReader.cs:252]: >position >= Data.Length [](http://example.com/codeflow?start=16&length=23)

is it possible to create a binary to test this cases?

Comment (shaopeng-gh) [reply src/BinaryParsers/ElfBinary/Dwarf/DwarfMemoryReader.cs:252]: yes! have no idea currently but let me check

Comment (shaopeng-gh) [reply src/BinaryParsers/ElfBinary/Dwarf/DwarfMemoryReader.cs:252]: not able to get the exact same issue but I found another issue which is similar out of boundary. it is when compiled with special machine type.
fixed the error and added 2 test files.

**Review by eddynaka — APPROVED** [review]

File: src/ReleaseHistory.md [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 ## Unreleased
 
+* BUGFIX: Fix exception System.AccessViolationException caused by trying to read data out of boundary. [#470](https://github.com/microsoft/binskim/pull/470)
```
Comment (eddynaka) [inline src/ReleaseHistory.md:5]:
```suggestion
* BUGFIX: Fix exception `System.AccessViolationException` caused by trying to read data out of boundary. [#470](https://github.com/microsoft/binskim/pull/470)
```

**Pull request merged**